src/main/java/com/google/firebase/messaging/AndroidFcmOptions.java

```diff
@@ -0,0 +1,67 @@
+package com.google.firebase.messaging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.Key;
+import com.google.firebase.internal.FcmOptionsUtil;
+
+/**
+ * Represents the Android-specific FCM options that can be included in an {@link AndroidConfig}.
+ * Instances of this class are thread-safe and immutable.
+ */
+public class AndroidFcmOptions {
```

Comment:
Lets also make this `final`. Same goes for all other public classes introduced in this PR.

Reply:
done